chia/consensus/multiprocess_validation.py

```diff
@@ -253,7 +261,8 @@ async def pre_validate_blocks_multiprocessing(
             final_pickled = recent_sb_compressed_pickled
         b_pickled: Optional[List[bytes]] = None
         hb_pickled: Optional[List[bytes]] = None
-        previous_generators: List[Optional[bytes]] = []
+        generators_bytes: List[Optional[bytes]] = []
+        prev_generator_args_bytes: List[Optional[List[Tuple[int, bytes]]]] = []
```

Comment:
I take it each element in `generator_bytes` is paired with the corresponding element in `prev_generator_args_bytes`. It seems it would make more sense to store them in a single list, together. Something like:

```
previous_generators_bytes: List[Tuple[Optional[Bytes], Optional[List[Tuple[int, bytes]]]]]
```

But maybe that gets too hairy

Reply:
In fact, it looks like the program and its args aren't individually `Optional`, but they are either both present or both missing. Som probably something like:

```
previous_generators_bytes: List[Optional[Tuple[Bytes, List[Tuple[int, bytes]]]]]
```

Reply:
Yeah that's a good idea, seems simpler and less error prone

Reply:
I think a comment explaining what's in there might make sense too, since it's a pretty complex structure